File: solutions/c/difference-of-squares/2/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number)
{
    return number * (number + 1) * (2 * number + 1) / 6;
}
unsigned int square_of_sum(unsigned int number)
{
    unsigned int result = number * (number + 1) / 2;
    return result * result;
}
unsigned int difference_of_squares(unsigned int number)
{
    return square_of_sum(number) - sum_of_squares(number);
}

File: solutions/ruby/difference-of-squares/2/difference_of_squares.rb
class Squares
  def initialize(n)
    @square_of_sum = (n * (n + 1) / 2)**2
    @sum_of_squares = n * (n + 1) * (2 * n + 1) / 6
  end

  def difference
    square_of_sum - sum_of_squares
  end

  attr_reader :square_of_sum
  attr_reader :sum_of_squares
end